### ISSUE TITLE
[Markdown] URLs do not highlight if line above contains a | (pipe) character and there is an empty line above it

### ISSUE THREAD

**Issue by rdela** [sublimehq/Packages#3154]
Sublime Text build 4122 does not syntax highlight URLs in Markdown files if the line above them contains a | (pipe) character and the line above (2 lines above the URL) is empty.

Normal behavior:

```markdown
Sublime Text 
https://www.sublimetext.com
```

Bug:

```markdown
Sublime Text | Text Editing, Done Right
https://www.sublimetext.com
```

Exception:

```markdown
The sophisticated text editor for code, markup and prose.
Sublime Text | Text Editing, Done Right
https://www.sublimetext.com
```

I created a [gist with the file I used to test this containing the examples above](https://gist.github.com/rdela/623227bed7f6bb26d02201a51bac4f44).

**Comment by deathaxe** [comment]
The pipe `|` causes the syntax to match the line as beginning of a table. It expects a table separator at the position of the url.

The issue is addressed in https://github.com/SublimeText-Markdown/MarkdownEditing/commit/836e3f63d681cf9a80777dea4c948da21b999b3b next to some other refactorings.

Note the commit is no final solution as it tries to keep ST3 compatible at this point.

A full fix will need branching to also prevent the first line from being matched as table header.

**Comment by deathaxe** [comment]
Fixed by PR #3167
